### PR TITLE
Return full pipfile_entry dict instead of cherry-picking values

### LINKED ISSUE #256: Key error raised on requirements subcommand when VCS dependency is specified
**Describe the bug**

Given the following Pipfile:

```toml
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
selinon = {git = "https://github.com/selinon/selinon"}

[dev-packages]
```

Running the following command:

```
micropipenv --verbose req --only-direct
```

I get the following error:

```python
Traceback (most recent call last):
  File "/Users/fridolin.pokorny/git/fridex/selinon/demo-worker/./micropipenv.py", line 1464, in <module>
    return_code = main()
  File "/Users/fridolin.pokorny/git/fridex/selinon/demo-worker/./micropipenv.py", line 1451, in main
    handler(**arguments)
  File "/Users/fridolin.pokorny/git/fridex/selinon/demo-worker/./micropipenv.py", line 1301, in requirements
    sections = sections or get_requirements_sections(
  File "/Users/fridolin.pokorny/git/fridex/selinon/demo-worker/./micropipenv.py", line 1097, in get_requirements_sections
    result["default"] = {
  File "/Users/fridolin.pokorny/git/fridex/selinon/demo-worker/./micropipenv.py", line 1098, in <dictcomp>
    dependency_name: _parse_pipfile_dependency_info(pipfile_entry)
  File "/Users/fridolin.pokorny/git/fridex/selinon/demo-worker/./micropipenv.py", line 1064, in _parse_pipfile_dependency_info
    result = {"version": pipfile_entry["version"]}
KeyError: 'version'
```

**Expected behavior**

Requirements should be generated respecting VCS dependency stated in the provided Pipfile.

### PR DESCRIPTION
**Pull request by VannTen**
## Related Issues and Dependencies
fix #256
…

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

If there is an unknown key, we won't use it in the rest of the code.
This way, we gain free support for dependencies using git file, etc.

## Caveat

I'm not completely sure of the full impact of the fix. But I think we only use specific keys in the returned dictionnary. Hence we should not have any problem with unknown ones, because we just won't use them.
But that also means we might accept invalid Pifile.

### CODE REVIEW

**Comment by frenzymadness** [comment]
I think this approach is fine. Could you please add some tests for it? You can at least transfer the bug report into a regression test so we'll have one more use case covered in tests.

**Comment by VannTen** [comment]
I plan to, I should do that this week.

**Comment by VannTen** [comment]
Looks like some pip version related warning, I suppose pip 23 is new.
I'll take a look, if time allows.

**Comment by frenzymadness** [comment]
Just change

https://github.com/thoth-station/micropipenv/blob/fe8c6fd6fc29b80d154dbf1d34f5266a477e9b89/micropipenv.py#L51

to `<=23.0`.

**Comment by VannTen** [comment]
/approve
I'll rebase on top of #258 once it's in.

**Review by frenzymadness — APPROVED** [review]
I like it and I've verified that the test failed with the previous version. Do you want to do a new release? You can just add a changelog, bump the version and push it to master with a corresponding tag.

**Comment by VannTen** [comment]
I'll do that, at least I'll know how the release part works.

**Comment by VannTen** [comment]
@frenzymadness can you put a /lgtm ?

**Review by frenzymadness — APPROVED** [review]

**Comment by frenzymadness** [comment]
I already approved it last week and I see no significant difference after the lgtm command.

**Comment by frenzymadness** [comment]
So, before the lgtm command was posted by me, I saw that I could merge this PR. Now, after the lgtm command, I cannot do that unless it passes all the checks or I use my admin privileges to bypass the rules for the protected branch. And the sesheta cheks will never finish I guess. I'm inclined to remove the bot again.

**Comment by VannTen** [comment]
looks like I don't have push right to master.
Btw, is the release to pypi handled by github as well ?

**Comment by frenzymadness** [comment]
> looks like I don't have push right to master. Btw, is the release to pypi handled by github as well ?

Yes, but it requires a tag which is not possible to do via PR so you can open a PR with a new version, changelog etc. and I'll merge it, tag it and push it to master which should lead to a new release on PyPI.

**Comment by VannTen** [comment]
And I had a local and signed tag :/. Oh well.

**Comment by frenzymadness** [comment]
So, do you want to be able to push directly to master?

**Comment by VannTen** [comment]
To do releases, yeah.

We could have some automation for releases and changelog, but I don't
have the time to invest in that right now, so for the time being let's
keep it manual.

**Pull request merged**
